core/primitives/src/errors.rs

```diff
@@ -332,6 +332,48 @@ pub struct ActionError {
     pub kind: ActionErrorKind,
 }
 
+#[derive(
+    BorshSerialize, BorshDeserialize, Debug, Clone, PartialEq, Eq, Deserialize, Serialize, RpcError,
+)]
+pub enum ContractCallError {
+    MethodResolveError(MethodResolveError),
+    CompilationError(CompilationError),
+    ExecutionError(String),
+}
+
+impl Into<FunctionCallErrorSer> for ContractCallError {
```

Comment:
@matklad Is my understanding correct that implementing `Into` is not rust-idiomatic? I think it is always preferred to implement `From`

Reply:
Oh, good catch @bowenwang1996! Yeah, this should go the other way arond, `From<ContractCallError> for FunctionCallErrorSer`.

Reply:
Replaced.